node/src/chain_spec/mod.rs

```diff
@@ -129,8 +130,101 @@ pub fn parachain_genesis(
 		// no need to pass anything, it will panic if we do. Session will take care
 		// of this.
 		aura: Default::default(),
+		asset_registry: AssetRegistryConfig {
+			asset_names: vec![],
+			native_asset_name: TOKEN_SYMBOL.as_bytes().to_vec(),
+			native_existential_deposit: NATIVE_EXISTENTIAL_DEPOSIT,
+		},
+		multi_transaction_payment: MultiTransactionPaymentConfig {
+			currencies: vec![],
+			fallback_account: Some(root_key),
+			account_currencies: vec![],
+		},
+		tokens: TokensConfig {
+			balances: if registered_assets.is_empty() {
+				vec![]
+			} else {
+				endowed_accounts
+					.iter()
+					.flat_map(|x| {
+						vec![
+							(x.0.clone(), 1, 1_000_000_000_000u128 * UNITS),
```

Comment:
This should be configurable based on params of the function. Aslo you can't use `alice`,`bob` or any "dev" constants. Everything in this function should be configurable from params because all the chains are calling this function to build chains spec. including `staging` or `testnet`

Reply:
Do you mean something like this?
```
pub fn parachain_genesis(
	wasm_binary: &[u8],
	root_key: AccountId,
	initial_authorities: Vec<(AccountId, AuraId)>,
	endowed_accounts: Vec<(AccountId, Balance)>,
	_enable_println: bool,
	parachain_id: ParaId,
	_council_members: Vec<AccountId>,
	_tech_committee_members: Vec<AccountId>,
	_tx_fee_payment_account: AccountId, // Account use multi-payment pallet to send fees to in pool does not exists
	_vesting_list: Vec<(AccountId, BlockNumber, BlockNumber, u32, Balance)>,
	registered_assets: Vec<(Vec<u8>, Balance)>, // (Asset name, Existential deposit)
	_accepted_assets: Vec<(AssetId, Price)>,     // (Asset id, Fallback price) - asset which fee can be paid with
	token_balances: Vec<(AccountId, Vec<(AssetId, Balance)>)>,          <-----------       HERE
) -> GenesisConfig {
```

Reply:
yes, it can be called `token_balances` but everything where  "constant" values (e.g `alice`, `bob` or `create_testnet_claims`) are used have to configurable